gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/plan/use_case/CreatePlanUseCase.java

```diff
@@ -83,7 +77,13 @@ private static boolean isMtls(Api api, Plan plan) {
         );
     }
 
-    public record Input(String apiId, Function<Api, Plan> toPlan, List<Flow> flows, AuditInfo auditInfo) {}
+    public record Input(
+        String apiId,
+        Function<Api, Plan> toPlan,
+        Supplier<List<Flow>> httpFlowProvider,
+        Supplier<List<NativeFlow>> nativeFlowProvider,
```

Comment:
what about replace Supplier to a Function With Api as args to merge this two args and lets the resource conditional the result ?

Reply:
of keep same logic with single `Supplier<? extends List<? extends AbstractFlow>>` ?

Reply:
I don't think that merging those two fields will give us any benefit. I think that it's better to make a call within logic, which already uses Api class.

Reply:
Don't you think it's better to keep the code consistent? it's the same need as here, the line above. you might as well use the same strategy, no?

https://github.com/gravitee-io/gravitee-api-management/blob/c59130ed15298c1caa73a4f717a55c8c9bb5b63a/gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/resource/api/ApiPlansResource.java#L181

https://github.com/gravitee-io/gravitee-api-management/blob/900e02c8f6f3142550a461a758e6ef31b9774d1c/gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/PlanMapper.java#L115

Reply:
Ok, thinking about consistency it makes sense